leakcanary-android-proguard/src/main/java/com/squareup/leakcanary/proguard/LeakcanaryProguardPlugin.kt

```diff
@@ -0,0 +1,42 @@
+package com.squareup.leakcanary.proguard
+
+import com.android.build.gradle.AppExtension
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+
+class LeakcanaryProguardPlugin : Plugin<Project> {
+  override fun apply(project: Project) {
+    val extension = project.extensions.create("leakCanary", LeakCanaryProguardExtension::class.java)
+
+    val androidPluginHandler = { _: Plugin<*> ->
+      val android = project.extensions.getByType(AppExtension::class.java)
+
+      project.afterEvaluate {
+        android.applicationVariants.all { variant ->
```

Comment:
Many plugins check that the android plugin is present. Here it seems to me it's worth it as the requirement is even stronger: you want only the app plugin. You could notify your users if they use it on a lib or a feature.

Reply:
I'm using this: https://docs.gradle.org/current/javadoc/org/gradle/api/plugins/PluginContainer.html#withId-java.lang.String-org.gradle.api.Action-

so if I understand this correctly if android plugin is not present then the whole `androidPluginHandler` block shouldn't be executed, right?

Reply:
I would just throw an error + message if the the application plugin is not present.

Reply:
I defined `androidPluginHandler` function in line 11 then I used it here:
https://github.com/square/leakcanary/blob/43b5b4c9b6318aed52c16aa6afd72849cbf36a95/leakcanary-android-proguard/src/main/java/com/squareup/leakcanary/proguard/LeakcanaryProguardPlugin.kt#L40

According to the doc this function should be called only if the plugin is present and I'm accessing android plugin only inside the function. Do you think having the additional check is still needed?

Reply:
I think @stephanenicolas meant that if someone used this on a library module, we should output an error to let them know it's a mistake.

Reply:
Libraries can run ProGuard and can have test APKs. Seems like it should be supported.

Also, don't call it ProGuard. It's minification. ProGuard is no longer the default minifier as of AGP 3.5

Reply:
I've always thought that libraries can only define their proguard rules files but the minification is not performed on a library itself but on the final apk. How should build.gradle file look for a library that I want to be minified?

Reply:
The same as an app. minifyEnabled true and specifying proguardFiles for rules.